File: Top-Interview-150/Two-Pointers/Is_Subsequence.py
# Problem: Is Subsequence
# Technique: Two Pointers
# Link: https://leetcode.com/problems/is-subsequence/description/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        pr_s = 0
        pr_t = 0

        while pr_t < len(t) and pr_s < len(s):
            if t[pr_t] == s[pr_s]:
                pr_s +=  1
            pr_t +=  1
        
        return pr_s == len(s)

File: Top-Interview-150/Two-Pointers/Is_Subsequence.rs
// Problem: Is Subsequence
// Technique: Two Pointers
// Link: https://leetcode.com/problems/is-subsequence/?envType=study-plan-v2&envId=top-interview-150

impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut s_chars = s.chars();
        let mut current = s_chars.next();

        for c in t.chars() {
            if let Some(ch) = current {
                if c == ch {
                    current = s_chars.next();
                }
            } else {
                break;
            }
        }

        current.is_none()
    }
}
